src/Operation/InsertMany.php

```diff
@@ -142,7 +153,14 @@ public function execute(Server $server)
         $insertedIds = [];
 
         foreach ($this->documents as $i => $document) {
-            $insertedIds[$i] = $bulk->insert($document);
+            $insertedDocument = isset($this->options['codec'])
+                ? $this->options['codec']->encodeIfSupported($document)
+                : $document;
+            // Psalm's assert-if-true annotation does not work with unions, so
+            // assert the type manually instead of using is_document
+            assert(is_array($insertedDocument) || is_object($insertedDocument));
+
+            $insertedIds[$i] = $bulk->insert($insertedDocument);
```

Comment:
Similar to my suggestion in FindOneAndReplace:

```suggestion
            if (isset($this->options['codec'])) {
                $document = $this->options['codec']->encodeIfSupported($document);
                // Psalm's assert-if-true annotation does not work with unions, so
                // assert the type manually instead of using is_document
                assert(is_array($document) || is_object($document));
            }
            
            $insertedIds[$i] = $bulk->insert($document);
```

Likewise in InsertOne below.

Reply:
Updated.